packages/swap/src/event-handlers/swapRequestCompleted.ts

```diff
@@ -35,29 +35,39 @@ export default async function swapRequestCompleted({
   if (
     channelQuote &&
     channelQuote?.expectedDepositAmount.toFixed(0) === swapRequest.depositAmount?.toFixed(0) &&
-    !channelQuote.swapRequestId && // do not overwrite data if channel is reused
-    !swapRequest.refundEgressId // only store execution price if full deposit amount was swapped
+    !channelQuote.swapRequestId // do not overwrite data if channel is reused
   ) {
-    const quotedPrice = channelQuote.quotedPrice.toNumber();
-    const executedPrice = Number(
-      getSwapPrice(
-        swapRequest.srcAsset,
-        swapRequest.swapInputAmount.toFixed(),
-        swapRequest.destAsset,
-        totalSwapOutputAmount.toFixed(),
-      ),
-    );
-    const executedSlippagePercent = (100 * (quotedPrice - executedPrice)) / quotedPrice;
+    if (swapRequest.refundEgressId) {
+      await prisma.quote.update({
+        where: { id: channelQuote.id },
+        data: {
+          refundedAt: new Date(block.timestamp),
+          swapRequestId: swapRequest.id,
+          swapRequestNativeId: swapRequest.nativeId,
+        },
+      });
+    } else {
+      const quotedPrice = channelQuote.quotedPrice.toNumber();
+      const executedPrice = Number(
+        getSwapPrice(
+          swapRequest.srcAsset,
+          swapRequest.swapInputAmount.toFixed(),
+          swapRequest.destAsset,
+          totalSwapOutputAmount.toFixed(),
+        ),
+      );
+      const executedSlippagePercent = (100 * (quotedPrice - executedPrice)) / quotedPrice;
 
-    await prisma.quote.update({
-      where: { id: channelQuote.id },
-      data: {
-        executedAt: new Date(block.timestamp),
-        executedPrice,
-        executedSlippagePercent,
-        swapRequestId: swapRequest.id,
-        swapRequestNativeId: swapRequest.nativeId,
-      },
-    });
+      await prisma.quote.update({
+        where: { id: channelQuote.id },
+        data: {
+          executedAt: new Date(block.timestamp),
+          executedPrice,
```

Comment:
we don't care about executed price if it was partially executed?

Reply:
i think it would be kinda misleading to store it for partially refunded swaps because its artifically limited by the tolerance of the swap - added more on this in https://github.com/chainflip-io/chainflip-sdk-monorepo/pull/1084#discussion_r1851890222